### PR TITLE
Package Upgrade: Refactors code to upgrade DiagnosticSource Library from 6.0.1 to 8.0.1 

### PR DESCRIPTION
**Pull request by sourabh1007**
### Why Upgrade?
The new `DiagnosticSource` library includes additional Meter API features that we plan to utilize moving forward.

### Is this a Breaking Change?
No, this upgrade is not expected to cause breaking changes. After reviewing this [documentation](https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/8.0/parameterinfo-hasdefaultvalue), no issues related to `DiagnosticSource` or `ActivitySource` were found.

### Potential Impact on Existing Customers?
Upon upgrading to this version, customers may need to update some dependencies that rely on the older version of the `DiagnosticSource` library.

## Type of change
- [] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj [hunk 1]
```diff
@@ -114,7 +114,7 @@
 	<ItemGroup>
 		<PackageReference Include="Azure.Core" Version="1.19.0" />
 		<PackageReference Include="System.Collections.Immutable" Version="1.7.0" />
-		<PackageReference Include="System.Diagnostics.DiagnosticSource" Version="6.0.1" />
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj:117]: Required package dependencies contract changes UT change as well.

Comment (sourabh1007) [reply Microsoft.Azure.Cosmos/src/Microsoft.Azure.Cosmos.csproj:117]: updated tests

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/DirectContractTests.cs [hunk 2]
```diff
@@ -177,10 +177,10 @@ public void PackageDependenciesTest()
                 }
             }
 
-            // Dependency version should match
+            // Dependency version should greater than minimum version defined
             foreach (KeyValuePair<string, Version> e in allDependencies)
             {
-                Assert.AreEqual(e.Value, projDependencies[e.Key], e.Key);
+                Assert.IsTrue(e.Value.CompareTo(projDependencies[e.Key]) <= 0, e.Key);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/DirectContractTests.cs:183]: Is this change needed?

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/DirectContractTests.cs:183]: Its the Hybridrow System.Memory that's violating lets pelase add specific expcetion.

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/DirectContractTests.cs:183]: Line 160 to below might suffice 

```C#
                    if (nuspecFile.Contains("hybridrow") && (e.Key.Contains("CompilerServices.Unsafe") || e.Key.Contains("System.Memory")))

```

Comment (sourabh1007) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/DirectContractTests.cs:183]: generally dependencies defined in these specifications are minimum version required...thats why I made this change if minimum version condition is satisfying then we should be good.

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/DirectContractTests.cs:183]: In theory that should do, unsure why we authored as exact to start with

**Review by kirankumarkolli — APPROVED** [review]
